net: nrf_cloud: DNS Improvements (#17104)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

      nRF Cloud still accepts ``lng`` for backward compatibility.
    * The handling of MQTT JITP device association to improve speed and reliability.
    * To use nRF Cloud's custom MQTT topics instead of the default AWS topics.

>     * MQTT and CoAP transports to use a single unified DNS lookup mechanism that supports IPv4 and IPv6, fallback to IPv4, and handling of multiple addresses returned by getaddrinfo.

Reviewer: Is getaddrinfo here a function, or the AT command? Regardless, it needs formatting.

>     * The :kconfig:option:`CONFIG_NRF_CLOUD_IPV6` Kconfig is now deprecated, and no longer forces the nRF Cloud MQTT transport to use IPv4 when not enabled.

Reviewer: ```suggestion

  * Deprecated the :kconfig:option:`CONFIG_NRF_CLOUD_IPV6` Kconfig, and it no longer forces the nRF Cloud MQTT transport to use IPv4 when not enabled.
```

>       Instead, use the :kconfig:option:`CONFIG_NET_IPV4` and :kconfig:option:`CONFIG_NET_IPV6` KConfig options to customize which IP versions the ``nrf_cloud`` library uses.

Reviewer: ```suggestion
    Instead, use the :kconfig:option:`CONFIG_NET_IPV4` and :kconfig:option:`CONFIG_NET_IPV6` Kconfig options to customize which IP versions the :ref:`lib_nrf_cloud` library uses.
```
